tests/tests_geomstats/test_gamma.py

```diff
@@ -0,0 +1,186 @@
+        geomstats.tests.tf_backend() or geomstats.tests.pytorch_backend()
+    )
+    skip_test_log_is_tangent = (
+        geomstats.tests.tf_backend() or geomstats.tests.pytorch_backend()
+    )
+    skip_test_dist_is_symmetric = (
+        geomstats.tests.tf_backend() or geomstats.tests.pytorch_backend()
+    )
+    skip_test_dist_is_positive = (
+        geomstats.tests.tf_backend() or geomstats.tests.pytorch_backend()
+    )
+    skip_test_squared_dist_is_symmetric = (
+        geomstats.tests.tf_backend() or geomstats.tests.pytorch_backend()
+    )
+    skip_test_squared_dist_is_positive = (
+        geomstats.tests.tf_backend() or geomstats.tests.pytorch_backend()
+    )
+    skip_test_dist_is_norm_of_log = (
+        geomstats.tests.tf_backend() or geomstats.tests.pytorch_backend()
+    )
+    skip_test_dist_point_to_itself_is_zero = (
+        geomstats.tests.tf_backend() or geomstats.tests.pytorch_backend()
+    )
+    skip_test_exp_after_log = True
+    skip_test_log_after_exp = True
+    skip_test_parallel_transport_ivp_is_isometry = True
+    skip_test_parallel_transport_bvp_is_isometry = True
+    skip_test_geodesic_ivp_belongs = True
+    skip_test_geodesic_bvp_belongs = True
+    skip_test_exp_geodesic_ivp = True
+    skip_test_exp_ladder_parallel_transport = True
+    skip_test_triangle_inequality_of_dist = (
+        geomstats.tests.tf_backend() or geomstats.tests.pytorch_backend()
+    )
+
+    testing_data = GammaMetricTestData()
+
+    @geomstats.tests.np_autograd_and_torch_only
+    def test_metric_matrix_shape(self, point, expected):
+        return self.assertAllClose(self.metric().metric_matrix(point).shape, expected)
+
+    @geomstats.tests.np_autograd_and_tf_only
+    def test_christoffels_vectorization(self, point, expected):
+        return self.assertAllClose(self.metric().christoffels(point), expected)
+
+    @geomstats.tests.np_autograd_and_tf_only
+    def test_christoffels_shape(self, point, expected):
+        return self.assertAllClose(
+            self.metric().christoffels(base_point=point).shape,
+            expected,
+        )
+
+    @geomstats.tests.np_and_autograd_only
+    def test_exp_vectorization(self, point, tangent_vecs, exp_solver):
+        """Test the case with one initial point and several tangent vectors."""
+        end_points = self.metric().exp(
+            tangent_vec=tangent_vecs, base_point=point, solver=exp_solver
+        )
+        result = end_points.shape
+        expected = (tangent_vecs.shape[0], 2)
+        self.assertAllClose(result, expected)
+
+    @geomstats.tests.np_and_autograd_only
+    def test_exp_control(self, base_point, tangent_vec, exp_solver):
+        """Test exp at a random base point for a tangent vector of controlled norm."""
+        end_point = self.metric().exp(
+            tangent_vec=tangent_vec, base_point=base_point, solver=exp_solver
+        )
+        result = gs.any(gs.isnan(end_point))
+        self.assertAllClose(result, False)
```

Comment:
Can we also test exp and log on points where the geodesic is known, or at least its shape ? e.g. if vertical lines are geodesics, then `exp` of a base point and a vertical vector should yield a point vertically aligned with the base point, and `log` of two vertically aligned points should yield a vertial vector.